### ISSUE TITLE
bintrees is discontinued in favor of sortedcontainers

### ISSUE THREAD

**Issue by afaerber** [angr/angr#947]
https://pypi.python.org/pypi/bintrees/2.0.7
> Bintrees Development Stopped
> Use sortedcontainers instead: https://pypi.python.org/pypi/sortedcontainers
> see also PyCon 2016 presentation: https://www.youtube.com/watch?v=7z2Ki44Vs4E
> Advantages:
> * pure Python no Cython/C dependencies
> * faster
> * active development
> * more & better testing/profiling

**Comment by afaerber** [comment]
File an issue against sortedcontainers requesting that function?

**Comment by zardus** [comment]
Worth trying for sure. Thanks for the suggestion.

I opened up https://github.com/grantjenks/sorted_containers/issues/87.

**Comment by zardus** [comment]
Looks like it's actually supported!

```
import sortedcontainers
s = sortedcontainers.SortedDict()
s[10] = 'ten'
s[3] = 'three'

# floor
assert next(s.irange(maximum=15, reverse=True)) == 10

# ceiling
assert next(s.irange(minimum=5, reverse=False)) == 10

# succimport sortedcontainers
s = sortedcontainers.SortedDict()
s[10] = 'ten'
s[3] = 'three'

# floor
assert next(s.irange(maximum=15, reverse=True)) == 10

# ceiling
assert next(s.irange(minimum=5, reverse=False)) == 10

# succ
assert next(s.irange(minimum=3, inclusive=(False, False), reverse=False)) == 10

# prev
assert next(s.irange(maximum=10, inclusive=(False, False), reverse=True)) == 3
assert next(s.irange(minimum=3, inclusive=(False, False), reverse=False)) == 10

# prev
assert next(s.irange(maximum=10, inclusive=(False, False), reverse=True)) == 3
```

Marking this as a "help wanted" pending someone having the time to switch this over :-)

**Comment by ltfish** [comment]
Addressed by PR #962. Close.
